ui/src/data_explorer/components/SendToDashboardOverlay.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {PureComponent} from 'react'
+
+import {getNewDashboardCell} from 'src/dashboards/utils/cellGetters'
+
+import {
+  OverlayContainer,
+  OverlayHeading,
+  OverlayBody,
+  MultiSelectDropdown,
+} from 'src/reusable_ui'
+
+import {addDashboardCellAsync} from 'src/dashboards/actions'
+
+import {QueryConfig, Dashboard, Source} from 'src/types'
+
+interface Props {
+  queryConfig: QueryConfig
+  dashboards: Dashboard[]
+  source: Source
+  rawText: string
+  onCancel: () => void
+  addDashboardCell: typeof addDashboardCellAsync
+}
+
+interface State {
+  selectedIDs: string[]
+  hasQuery: boolean
+  name: string
+}
+
+class SendToDashboardOverlay extends PureComponent<Props, State> {
+  constructor(props) {
+    super(props)
+    const {queryConfig} = this.props
+
+    this.state = {
+      selectedIDs: [],
+      hasQuery: queryConfig.fields.length !== 0,
```

Comment:
since `fields` is optional on a `QueryConfig`, it would be good to guard this property access with
```
  hasQuery: queryConfig.fields && queryConfig.fields.length !== 0
```

Reply:
or via `getDeep`, if that's your style

Reply:
we could also just check if rawText (the query) is an empty string or if queryConfig is null. thoughts?